website/server/controllers/api-v3/tasks.js

```diff
@@ -1215,4 +1242,52 @@ api.deleteTask = {
   },
 };
 
+/**
+ * @api {post} /api/v3/tasks/:taskId/undo Undo scoring of a task given its id
+ * @apiName UndoTask
+ * @apiGroup Task
+ *
+ * @apiParam {String} taskId The task _id or alias
+ *
+ * @apiExample {json} Example call:
+ * curl -X "POST" https://habitica.com/api/v3/tasks/3d5d324d-a042-4d5f-872e-0553e228553e/undo
+ *
+ * @apiSuccess {Object} data An empty object
+ *
+ * @apiUse TaskNotFound
+ */
+api.undoTask = {
+  method: 'POST',
+  url: '/tasks/:taskId/undo',
+  middlewares: [authWithHeaders()],
+  async handler (req, res) {
+    let user = res.locals.user;
+    let taskId = req.params.taskId;
+    let task = await Tasks.Task.findByIdOrAlias(taskId, user._id);
+
+    if (!task) {
+      throw new NotFound(res.t('taskNotFound'));
+    }
+
+    let latestHistory = task.history[task.history.length - 1];
+    user.stats.hp -= latestHistory.stats.hp;
+    user.stats.mp -= latestHistory.stats.mp;
+    user.stats.exp -= latestHistory.stats.exp;
+    user.stats.gp -= latestHistory.stats.gp;
+    if (latestHistory.quest && latestHistory.quest.progressDelta) user.party.quest.progress.up -= latestHistory.quest.progressDelta;
```

Comment:
Hmm. There is much to consider if we keeps these undos for a long time. I think if now we reset undos every time the user refreshes we can prevent worrying about this. 

Or, I can check if the user quest still exists.